src/misc.jl

```diff
@@ -201,3 +201,55 @@ end
     root --> :top
     GraphPlot((source, destiny))
 end
+
+
+@userplot AndrewsPlot
+
+"""
+    andrewsplot(data::AbstractArray; kind=[])
+
+https://en.wikipedia.org/wiki/Andrews_plot
+
+Example
+```julia
+using RDatasets, PlotRecipes
+iris   = dataset("datasets", "iris")
+y      = Array(iris[:,1:4])
+kind   = Array(iris[:,5])
+andrewsplot(y, kind=kind)
+```
+"""
+andrewsplot
+
+@recipe function f(h::AndrewsPlot; kind=[])
+    y = h.args[1]
```

Comment:
You could check whether `y` is a DataFrame (it's a dependency through StatPlots) and convert to Matrix{Float} . If you use the DataArray convert function you can replace any NA's with NaN, which can be handled by Plots.